zpay32/invoice_internal_test.go

```diff
@@ -230,13 +302,504 @@ func TestParseTimestamp(t *testing.T) {
 	for i, test := range tests {
 		time, err := parseTimestamp(test.data)
 		if (err == nil) != test.valid {
-			t.Errorf("Data decoding test %d failed: %v", i, err)
+			t.Errorf("timestamp decoding test %d failed: %v", i, err)
 			return
 		}
 		if test.valid && time != test.result {
-			t.Errorf("Timestamp decoding test %d failed: expected "+
-				"timestamp %d, got %d", i, test.result, time)
+			t.Errorf("%d) failed decoding timestamp: "+
+				"expected %d, got %d",
+				i, test.result, time)
+			return
+		}
+	}
+}
+
+// TestParseFieldDataLength checks that the 16 bit length is properly parsed.
+func TestParseFieldDataLength(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		data   []byte
+		valid  bool
+		result uint16
+	}{
+		{
+			data:  []byte{},
+			valid: false, // empty data
+		},
+		{
+			data:  []byte{0x0},
+			valid: false, // data too short
+		},
+		{
+			data:  []byte{0x0, 0x0, 0x0},
+			valid: false, // data too long
+		},
+		{
+			data:   []byte{0x0, 0x0},
+			valid:  true,
+			result: 0,
+		},
+		{
+			data:   []byte{0xf, 0xf},
+			valid:  true,
+			result: 495,
```

Comment:
max value is `{0x1F, 0x1F} == {0b11111, 0b11111}` since we are using `base32`. This data is actually invalid :P